### ISSUE TITLE
feature_importance question 

### ISSUE THREAD

**Issue by asheetal** [ModelOriented/DALEX#232]
I am working on a research paper. The reviewers have requested to furnish feature importance on unseen dataset. Is there a guidance on whether the feature importance should be run on unseen dataset? Should the feature_importance between training data and unseen data match or not guaranteed?

**Comment by pbiecek** [comment]
Practices are different. In the case of linear models, the variable importance  is usually assessed on the training data. In the case of random forests, an internal out-of-bag sample is used which indirectly corresponds to the unseen data through the tree. In the case of boosting, by default, the importance of the variables is assessed on a trained model without external validation data. In the original paper of Rudin FI scores were calculated on the training data (but on all permutations so it is also a bit different than in our case). 

As usual, it depends on the goal. If the model is not overfitted then the choice of the validation set should have small impact of FI scores. 

In DALEX, the feature importance is calculated by default on the data set given in the data argument in the explain() function. You can put there the training data as well as the validation data.

**Comment by pbiecek** [comment]
Having thought through the matter, I don't see any arguments for favouring feature importance on a teaching or test data. 

The performance on the test set will be lower on average, but when it comes to the importance of variables, the choice of the set for permutation should not matter.

In goodness-of-fit problems, it is easier to use a train set.
For goodness-of-prediction problems, you can use the test set.

If the result on the train set is different from the test set, it is worth checking what might be the cause. It may be that the model was overfitted.
